### PR TITLE
Driver health UX improvements

### PR DESCRIPTION
**Pull request by chelseakomlo**
The resulting output is:

`nomad node-status -self`
```
Drivers                   
qemu rkt docker (unhealthy) exec raw_exec java  
```
and

`nomad node-status --verbose -self`
```
Drivers                                             
Driver    Detected  Healthy  Message                        Time
docker    false     false    Driver docker is not detected  2018-04-03T17:35:20Z
exec      true      true     <none>                         2018-04-03T17:32:20Z
java      true      true     <none>                         2018-04-03T17:32:20Z
qemu      true      true     <none>                         2018-04-03T17:32:20Z
raw_exec  true      true     <none>                         2018-04-03T17:32:20Z
rkt       true      true     <none>                         2018-04-03T17:32:20Z
```

### CODE REVIEW

File: command/node_status.go [hunk 1]
```diff
@@ -384,17 +388,27 @@ func (c *NodeStatusCommand) formatNode(client *api.Client, node *api.Node) int {
...
 
+func (c *NodeStatusCommand) outputTruncatedNodeDriverInfo(node *api.Node) {
+	c.Ui.Output(c.Colorize().Color("\n[bold]Drivers"))
+	drivers := make([]string, 1)
+
+	for driverName, driverInfo := range node.Drivers {
+		if !driverInfo.Healthy {
+			drivers = append(drivers, fmt.Sprintf("%s (unhealthy)", driverName))
+		} else {
+			drivers = append(drivers, driverName)
```
Comment (schmichael) [inline command/node_status.go]: I think we should add a comma to make it clear which driver `(unhealthy)` applies to.

File: command/node_status.go [hunk 2]
```diff
@@ -325,7 +325,11 @@ func (c *NodeStatusCommand) formatNode(client *api.Client, node *api.Node) int {
 		c.Ui.Output(c.Colorize().Color(formatKV(basic)))
 
 		// Emit the driver info
-		c.outputNodeDriverInfo(node)
+		if c.verbose {
+			c.outputNodeDriverInfo(node)
+		} else {
+			c.outputTruncatedNodeDriverInfo(node)
```
Comment (dadgar) [inline command/node_status.go]: In the non-verbose case can we just add the output to the basic k/v list: https://github.com/hashicorp/nomad/pull/4098/files#diff-f2914d6bff3d2f22f80da63d7003d9fdR308

**Comment by chelseakomlo** [comment]
Thanks for the great feedback! 

Output now looks like: 

`nomad node-status -self`
```
ID            = 667d207a
Name          = linux
Class         = <none>
DC            = dc1
Drain         = false
Eligibility   = eligible
Status        = ready
Uptime        = 2h0m0s
Driver Status = java, qemu, rkt, docker (unhealthy), exec, raw_exec
```

and 
`nomad node-status --verbose -self`
```
Drivers                                             
Driver    Detected  Healthy  Message  Time          
docker    true      true     <none>   2018-04-05T18:28:27Z
exec      true      true     <none>   2018-04-05T18:28:27Z
java      true      true     <none>   2018-04-05T18:28:27Z                                                                                                                                                         qemu      true      true     <none>   2018-04-05T18:28:27Z
raw_exec  true      true     <none>   2018-04-05T18:28:27Z
rkt       true      true     <none>   2018-04-05T18:28:27Z
                                                                                                
```

File: command/node_status.go [hunk 3]
```diff
@@ -384,17 +392,26 @@ func (c *NodeStatusCommand) formatNode(client *api.Client, node *api.Node) int {
 
 }
 
+func (c *NodeStatusCommand) outputTruncatedNodeDriverInfo(node *api.Node) string {
+	drivers := make([]string, 1)
```
Comment (dadgar) [inline command/node_status.go]: Make the length of the initialized list equal to `len(node.Drivers`. Also it should be `make([]string, 0, len(node.Drivers))`. You are setting the initial length to 1 and then appending to it so index 0 here is empty

File: command/node_status.go [hunk 4]
```diff
@@ -384,17 +392,26 @@ func (c *NodeStatusCommand) formatNode(client *api.Client, node *api.Node) int {
...
+
+	for driverName, driverInfo := range node.Drivers {
+		if !driverInfo.Healthy {
+			drivers = append(drivers, fmt.Sprintf("%s (unhealthy)", driverName))
+		} else {
+			drivers = append(drivers, driverName)
+		}
+	}
+	return strings.Trim(strings.Join(drivers, ", "), ", ")
```
Comment (dadgar) [inline command/node_status.go]: Why the trim. Join won't add one to the last element.

Comment (chelseakomlo) [reply command/node_status.go]: Not sure I understand- Trim replaces all leading and trailing characters: https://golang.org/pkg/strings/#Trim

**Pull request merged**